lib/utils/validateTypes.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/**
+ * Checks if the value is a boolean or a Boolean object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isBoolean(value) {
+	return typeof value === 'boolean' || value instanceof Boolean;
+}
+
+/**
+ * Checks if the value is a number or a Number object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isNumber(value) {
+	return typeof value === 'number' || value instanceof Number;
+}
+
+/**
+ * Checks if the value is a plain object.
+ * @param {any} value
+ * @returns {boolean}
+ */
+function isPlainObject(value) {
```

Comment:
Here right implementation https://github.com/jonschlinkert/is-plain-object/blob/master/is-plain-object.js

Reply:
I think the simple check is okay.

Reply:
@alexander-akait Thanks for sharing it. I also thought the simple check is enough, but implementing `isPlainObject` seems more challenging than I thought.

So, I try the [`is-plain-object` package's test cases](https://github.com/jonschlinkert/is-plain-object/blob/0a47f0f6cd10e0d2489beb55a32a8d0ba7b04b25/test/server.js) also in our codebase:

```js
// lib/isPlainObject.test.js
const assert = require('assert');
const { isPlainObject } = require('./utils/validateTypes');

test('isPlainObject', () => {
	assert(isPlainObject(Object.create({})));
	assert(isPlainObject(Object.create(Object.prototype)));
	assert(isPlainObject({ foo: 'bar' }));
	assert(isPlainObject({}));
	assert(isPlainObject(Object.create(null)));

	function Foo() {
		this.abc = {};
	}

	assert(!isPlainObject(/foo/));
	assert(!isPlainObject(function () {}));
	assert(!isPlainObject(1));
	assert(!isPlainObject(['foo', 'bar']));
	assert(!isPlainObject([]));
	assert(!isPlainObject(new Foo()));
	assert(!isPlainObject(null));
});
```

As a result, the test fails:

```console
$ npx jest lib/isPlainObject.test.js
 FAIL  lib/isPlainObject.test.js
  ✕ isPlainObject (8 ms)

  ● isPlainObject

    assert(received)

    Expected value to be equal to:
      true
    Received:
      false

      18 | 	assert(!isPlainObject(['foo', 'bar']));
      19 | 	assert(!isPlainObject([]));
    > 20 | 	assert(!isPlainObject(new Foo()));
         | 	^
      21 | 	assert(!isPlainObject(null));
      22 | });
      23 |

      at Object.<anonymous> (lib/isPlainObject.test.js:20:2)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.692 s, estimated 1 s
Ran all test suites matching /lib\/isPlainObject.test.js/i.
```

Thus, I think it seems better to install [`is-plain-object`](https://www.npmjs.com/package/is-plain-object) for compatibility. Fortunately, that package supports ESM, and has no dependencies and its type definition also.

What do you think?

Reply:
Thank you @alexander-akait and @ybiquitous for investigating! I also didn't realize it would be challenging.

Given this, I think it is a good idea to install `is-plain-object`. This is done in 8ec74f6c.